### DISCUSSION TITLE
How to use vless+ws+tls to proxy

### DISCUSSION THREAD

**Discussion by GrowingHacker** [betalgo/openai#234] [Q&A] [answered]
I try to use normal http proxy,It's ok :
var handler = new HttpClientHandler()
            {
                Proxy = new WebProxy("192.168.0.103", 1082),
                UseProxy = true
            };

            var client = new HttpClient(handler);

            var openAiService = new OpenAIService(new OpenAiOptions()
            {
                ApiKey =myKeys,
            },client);

But my proxy server uses vless+ws+tls,How do i rewrite the code

**Comment by Frogley** [comment, marked as answer]
I guess your requirement is to use a V2Ray proxy within HttpClientHandler. The easiest way is to run a V2Ray client locally, which will provide a local HTTP proxy, and then just use this HTTP proxy directly. If you want to connect to the V2Ray server directly in HttpClientHandler without using the local HTTP proxy, it's a challenging task, because the HttpClientHandler class is mainly used for handling HTTP proxies, while the V2Ray is not a standard HTTP proxy. Although it is theoretically possible, it requires a certain degree of modification or inheritance of the HttpClientHandler class, as well as an in-depth understanding of the V2Ray communication protocol.

**Reply by GrowingHacker** [reply]
实际上我已经用v2ray核心直接进行本地http代理了，昨天研究到凌晨两点才发现v2ray核心本身只要修改json配置文件就可以工作了，而v2rayN等软件只是起到一个修改配置和选择核心的一个管理工具罢了。
In fact, I have used the v2ray core to directly perform local http proxy, and yesterday I researched until two o'clock in the morning to find that the v2ray core itself can work by modifying the json configuration file, while software such as v2rayN is just a management tool to modify the configuration and select the core.
